### DISCUSSION TITLE
PL Sketch Feature / Element

### DISCUSSION THREAD

**Discussion by eliotwrobson** [PrairieLearn/PrairieLearn#7797] [Ideas]
Continuation of discussion on this PR: https://github.com/PrairieLearn/PrairieLearn/pull/2686

cc @mfsilva22 @nwalters512

**Comment by mfsilva22** [comment]
I don't have specific requirements for this feature, but I believe adding a sketch area will be very beneficial to some courses. Adding it as an element would be beneficial, in case some may want to assign points to the sketch.

**Comment by firasm** [comment]
I brainstormed some ideas for `pl-sketch` ; some of these are already possible with the existing element (in #8390).

- [x] Multiple sketches per question
- [x] Multiple Colours
- [x] Multiple line thicknesses
- [x] Basic Shapes (line, square, circle, triangle)
- [x] Export sketch to png/pdf
- [x] Provide one or more images as a background (template) and students draw on top of the image. Example: incomplete Kreb's cycle
- [ ] Students drag and drop images on the canvas
- [x] Clear canvas screen
- [ ] (Advanced!) Instructor annotations as another "Layer" on top of student submissions
- [ ] (Advanced) Some type of accessibility support

**Reply by firasm** [reply]
Most of this is now possible with pl-excalidraw!

https://docs.prairielearn.com/elements/pl-excalidraw/

**Comment by reteps** [comment]
More info: https://github.com/PrairieLearn/PrairieLearn/pull/13750
